internal/k8s/reconciler/deployer.go

```diff
@@ -52,6 +54,10 @@ func (d *GatewayDeployer) Deploy(ctx context.Context, gateway *K8sGateway) error
 		return err
 	}
 
+	if err := d.ensureSecret(ctx, gateway.Gateway); err != nil {
```

Comment:
You can see here that we create the `Secret` containing the CA Cert _before_ we create the `Deployment` that needs to reference the `Secret` for the contents of one of its volumes.